### PR TITLE
Fix battle timer to display wall-clock seconds

### PR DESCRIPTION
**Pull request by ncipollo**
The battle turn timer in the status bar was displaying raw tick counts rather than elapsed seconds. At the default 1000 ms/tick rate this was coincidentally correct, but at any other rate the display was wrong — a 30-second timer at 500 ms/tick showed "60" instead of "30". This converts the timer to wall-clock seconds before it ever leaves the server.

- Renamed `countdown_ticks`/`max_turn_ticks` to `countdown_secs`/`max_turn_secs` throughout the message chain to make the unit clear at every layer
- Tick-to-second conversion added at the two origin points in `battle/tick.rs` and `room_threats.rs` using `ticks * tick_rate_ms / 1000`, with a `.max(1)` guard against division by zero
- TUI render in `battle/render.rs` now reads already-correct seconds directly, requiring no client-side conversion logic
- Three unit tests added verifying correctness at 1000 ms/tick, 500 ms/tick, and tick-rate-independence across four different rates

<details>
<summary>Agent Context</summary>

Closes #130. Related to #120 (turn timer countdown feature that introduced the raw-tick display).

**Root cause:** `BattleSnapshot.countdown_ticks` held a raw tick count. The TUI rendered it verbatim. At `tick_rate_ms = 1000` this looked correct (1 tick = 1 s) but was wrong at any other rate.

**Fix approach:** Rename fields to `_secs` and convert at the two server-side origin points that already have access to `game_state.mud_config.game_loop.tick_rate_ms`. The alternative (passing `tick_rate_ms` to the TUI and converting client-side) was rejected because it leaks internal game loop config into the presentation layer.

**Files changed:**
- `src/game/messaging.rs` — `BattleUpdateMessage` and `BattleStartedMessage` field rename
- `src/network/event.rs` — `BattleSnapshot` field rename
- `src/game/engagement/battle/tick.rs` — conversion + `BattleUpdateParams` rename + unit tests
- `src/game/interaction/room_threats.rs` — conversion in `build_battle_started_message`
- `src/network/server/message_relay.rs` — field rename in `battle_update_snapshot` and `battle_snapshot_from_message`
- `src/tui/app/battle_state.rs` — field rename in test helper `make_snapshot`
- `src/tui/app/network_event_handler.rs` — field rename in snapshot update handler
- `src/tui/screens/battle/render.rs` — field rename in `render_status_bar`

**Conversion formula:** `secs = ticks * tick_rate_ms.max(1) / 1000`

**Edge cases:** `tick_rate_ms = 0` guarded with `.max(1)`; progress bar denominator uses `.max(1)` (already present) to avoid division by zero.

**Tests:** `timer_converts_correctly_at_1000ms_tick_rate`, `timer_converts_correctly_at_500ms_tick_rate`, `timer_display_is_tick_rate_independent` (checks 250/500/1000/2000 ms rates all produce 30 s for a 30 000 ms window).
</details>

### CODE REVIEW

File: src/game/engagement/battle/tick.rs [hunk 1]
```diff
@@ -91,14 +91,17 @@ async fn handle_tick(
     let player_ids = find_participant_player_ids(game_state, &all_ids).await;
 
     let countdown_ticks = max_engage_ticks.saturating_sub(result.ticks_in_phase);
+    let tick_rate_ms = game_state.mud_config.game_loop.tick_rate_ms.max(1);
+    let countdown_secs = countdown_ticks * tick_rate_ms / 1000;
+    let max_turn_secs = max_engage_ticks * tick_rate_ms / 1000;
```
Comment (ncipollo) [inline src/game/engagement/battle/tick.rs]: let's extract these and the tests associated with the battle timer into their own file.
